draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1873,17 +1873,88 @@ still provide adequate output for incomplete logs.
+
+The simplest and most extreme form of protection against abuse of this
+information is the complete deletion of a given field, which is equivalent to
+not logging the field(s) in question. While deletion completely protects the
+data in the deleted fields from the risk of compromise, it also reduces the
+utility of the dataset as a whole. As such, a balance should be found between
+logging these fields and the potential risks inherent in their (involuntary)
+disclosure. This balance depends on the use case at hand (e.g., research
+datasets have different requirements to live operational troubleshooting). As
+such, operators SHOULD only capture the minimal subset of qlog data relevant to
+their specific purpose. Consequently, qlog implementers SHOULD allow for
+changing this subset in a dynamic fashion, ideally even on a per-connection
+basis.
+
+For the remaining fields deemed necessary for the use case at hand, as per
+{{!RFC6973}}, operators must be aware that the data will be at risk of
+compromise. As such, measures should be taken to firstly reduce the risk of
+compromise and secondly reduce the risk of abuse of compromised data. While a
+full discussion of both aspects is out of scope for this document, the following
+paragraphs discuss high level concepts that can be applied to qlog data.
+
+To reduce the risk of compromise, operators can take measures such as: limiting
+the amount of time data is stored, encrypting data at rest, limiting access
+rights to the data, and auditing data usage practices. qlog deployments SHOULD
+have integrated options for easy qlog data deletion or (aggressive) aggregation.
+
+To reduce the risk of data abuse after compromise, sensitive fields can be
+anonymized, pseudonymized, otherwise permutated/replaced, truncated,
+(re-)encrypted, or aggregated. A partial discussion of applicable techniques
+(especially for IP address information) can be found in Appendix B of
+{{!DNS-PRIVACY=RFC8932}}. Operators should however be aware that many of these
+techniques have been shown to be insufficient to safeguard user privacy and/or
+to protect user identity, especially if qlog data would be compromised in large
+amounts or be correlated with other data sources. As such, techniques from the
+field of differential privacy (TODO: add survey paper reference) should be
+applied to mitigate this risk to its maximal potential.
+
+Finally, in keeping with the trend of maximum end user agency, qlog operators
+SHOULD take end user preferences into account. While active user participation
+(as per {{!RFC6973}}) on a per-qlog basis is difficult, as logs are often capture
+out-of-band to the main user interaction and intent, implementers should have a
+mechanism for integrating the capture, storage and removal of qlogs with more
+general, often existing, user consent and privacy control systems. This is
+especially true when logging (plaintext) packet and frame payload, as this can
+contain highly sensitive information.
```

Comment:
I find this one a little hard to align with as written. what is the trend? Will that trend be one in 10 years time?

It might be better to rephrase the start is in terms of qlog operators needing to consider something we didn't alread mention: then tention between their operational needs and end user rights or preferences (the rights is important here, such as cookie laws, GDPR, regional policies etc). We can then close out the paragraph information that might contain more information that is sensitive or confidential to the end user (where previously we talked abput sensitivity mainly through the lens of connections), and then use the packet or frame payloads as example of where that data might reside.